### ISSUE TITLE
[BUG] While conducting mobile searches, a timeout error occurs after a certain number of searches.

### ISSUE THREAD

**Issue by Chappie18** [charlesbel/Microsoft-Rewards-Farmer#277]
### Before submitting a bug report...

- [X] This bug wasn't already reported.
(I have checked every bug report on GitHub)

### Title

- [X] The title is no longer "[BUG] Title" and I edited it with the right error name.

### Describe the bug

While doing mobile searches, a timeout error appears after a few searches, and I also checked with another account and this error appears. I also reconnected my VPN, thinking it was a network issue, but the error still appears after 4 or 5 searches, just on mobile searches.

### Copy and paste your error

[ERROR] [BING] Timeout, retrying in 5 seconds...

### Screenshots

![image](https://github.com/charlesbel/Microsoft-Rewards-Farmer/assets/67442659/3cd765bf-b9e8-4cd4-9e56-4bdc535b118a)

### Value of dashboard variable

![Screenshot 2023-08-15 033900](https://github.com/charlesbel/Microsoft-Rewards-Farmer/assets/67442659/c54470b9-a9cb-4237-9296-7b18ef04022b)

**Comment by Hirschkaeferkind** [comment]
After a few retrys, the script should skip this part and move on to the next step. Maybe an extra switch for just doing the mobile searches? Just an idea.

**Comment by SgtSobieski** [comment]
Could it be an error because of the proxy? I got it also and I am using socks5.

**Comment by Godielvs** [comment]
im getting this constantly and im unable to continue using it. it just stays at "retrying in x seconds" for hours and hours

**Comment by Chappie18** [comment]
> im getting this constantly and im unable to continue using it. it just stays at "retrying in x seconds" for hours and hours

I have been experiencing the same issue for the past several days. If there is an i am also facing same for past days here, if there is an argument to disable Mobile Searches or any other activities in addition to **DailySet, MorePromotions, PunchCards, and Searches**, it will be helpful in an emergency.

![image](https://github.com/charlesbel/Microsoft-Rewards-Farmer/assets/67442659/114705ee-64d4-4c75-879c-ad48975bb8f3)

**Comment by MaskBoyz** [comment]
Its only happen when doing mobile searches and never able to load bing.com again until I restart the script and after I restart then  after completing 5-15 searches its have the same issue

**Comment by theskid31** [comment]
yes happening to me only on mobile searches ,only after updateing undetected chrome to jdlocks version of 116
jdholtz:115-compatibility

**Comment by theskid31** [comment]
> Its only happen when doing mobile searches and never able to load bing.com again until I restart the script and after I restart then after completing 5-15 searches its have the same issue,
> 
> For me this happening after the update of Undetected Chrome

where did you get undected chrome update from
or did you run pip install undetected-chrome --update?

**Comment by theskid31** [comment]
> After a few retrys, the script should skip this part and move on to the next step. Maybe an extra switch for just doing the mobile searches? Just an idea.

yes we need a switch to stop this after so many continous retries so we can start again if required in a tasked schedule

**Comment by udu3324** [comment]
Modified something and finally got full 20 mobile searches, but timeout still happens?
![image](https://github.com/charlesbel/Microsoft-Rewards-Farmer/assets/47045986/28dfc144-3ea0-47f2-9066-536be3c7833e)

**Comment by theskid31** [comment]
> Modified something and finally got full 20 mobile searches, but timeout still happens? ![image](https://user-images.githubusercontent.com/47045986/261814410-28dfc144-3ea0-47f2-9066-536be3c7833e.png)

what did you modify?

**Comment by udu3324** [comment]
> > Modified something and finally got full 20 mobile searches, but timeout still happens? ![image](https://user-images.githubusercontent.com/47045986/261814410-28dfc144-3ea0-47f2-9066-536be3c7833e.png)
> 
> what did you modify?

I made it so it only goes to bing one time and returns after 5 timeout exceptions

**Comment by theskid31** [comment]
> > > Modified something and finally got full 20 mobile searches, but timeout still happens? ![image](https://user-images.githubusercontent.com/47045986/261814410-28dfc144-3ea0-47f2-9066-536be3c7833e.png)
> > 
> > 
> > what did you modify?
> 
> I made it so it only goes to bing one time and returns after 5 timeout exceptions
> 
> In #276

any chance you can share your altered version as i'm not that good at python scripting yet
and it would help me learn a bit ,thanks

**Comment by udu3324** [comment]
> Is there any effective solution now? I can hardly use the script because there are just too many timeout :(
> it took so long but only got few reward points(like < 40 per account each run)

Use the fork ive created #276

**Comment by tony11306** [comment]
> > Is there any effective solution now? I can hardly use the script because there are just too many timeout :(
> > it took so long but only got few reward points(like < 40 per account each run)
> 
> Use the fork ive created #276

I did, but I still got that timeout issue going on. 

Also that CPU fix in your fork did not work for me either. There's a `__exit__` function in Browser class, and I've tested the `driver.quit()` in it does execute when leaving the `with` statement, but after `driver.quit()` there were still some chrome processes running in the background. I think this might be chromedriver's or selenium's problem. The only solution I can think about now is to eliminate all chrome.exe each run, but that also implies you basically can't use chrome during execution.

**Comment by udu3324** [comment]
> > > Is there any effective solution now? I can hardly use the script because there are just too many timeout :(
> > > it took so long but only got few reward points(like < 40 per account each run)
> > 
> > Use the fork ive created #276
> 
> I did, but I still got that timeout issue going on. 
> 
> Also that CPU fix in your fork did not work for me either. There's a `__exit__` function in Browser class, and I've tested the `driver.quit()` in it does execute when leaving the `with` statement, but after `driver.quit()` there were still some chrome processes running in the background. I think this might be chromedriver's or selenium's problem. The only solution I can think about now is to eliminate all chrome.exe each run, but that also implies you basically can't use chrome during execution.

That's strange. I use the fork and the timeout issue is fixed for me.

**Comment by tony11306** [comment]
> > > > Is there any effective solution now? I can hardly use the script because there are just too many timeout :(
> > > > it took so long but only got few reward points(like < 40 per account each run)
> > > 
> > > 
> > > Use the fork ive created #276
> > 
> > 
> > I did, but I still got that timeout issue going on.
> > Also that CPU fix in your fork did not work for me either. There's a `__exit__` function in Browser class, and I've tested the `driver.quit()` in it does execute when leaving the `with` statement, but after `driver.quit()` there were still some chrome processes running in the background. I think this might be chromedriver's or selenium's problem. The only solution I can think about now is to eliminate all chrome.exe each run, but that also implies you basically can't use chrome during execution.
> 
> That's strange. I use the fork and the timeout issue is fixed for me.

idk it could be that I didn't delete the sessions folder after the update. I will delete it and try that again tomorrow.

**Comment by Godielvs** [comment]
I just made a .bat file that kills all chrome instances then executes main.py, and executes 5 times. Yeah you kinda can't use Chrome so I just open when I'm leaving the pc

**Comment by theskid31** [comment]
i implemented subprocess to kill chrome.exe
you may need to install subprocess : pip install subprocees

[https://github.com/theskid31/Microsoft-Rewards-Farmer.git](url)

**Comment by tony11306** [comment]
Yap, it was the sessions folder. I deleted and it worked a lot more better now. Thank you @udu3324
